nautobot/extras/factory.py

```diff
@@ -0,0 +1,61 @@
+from django.contrib.contenttypes.models import ContentType
+from django.utils.text import slugify
+
+import factory
+import faker
+
+from nautobot.core.factory import OrganizationalModelFactory
+from nautobot.extras.models import Status, Tag
+from nautobot.extras.utils import FeatureQuery, TaggableClassesQuery
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.factory import get_random_instances
+
+
+# Status isn't technically an OrganizationalModel, but it has all of its features **except** dynamic-groups
+class StatusFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Status
+        exclude = ("has_description",)
+
+    name = factory.LazyFunction(
+        lambda: "".join(word.title() for word in faker.Faker().words(nb=2, part_of_speech="adjective", unique=True))
+    )
+    slug = factory.LazyAttribute(lambda tag: slugify(tag.name))
+    color = factory.Iterator(ColorChoices.CHOICES, getter=lambda choice: choice[0])
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+    @factory.post_generation
+    def content_types(self, create, extracted, **kwargs):
+        if create:
+            if extracted:
+                self.content_types.set(extracted)
+            else:
+                self.content_types.set(
+                    get_random_instances(
+                        lambda: ContentType.objects.filter(FeatureQuery("statuses").get_query()), minimum=1
+                    )
+                )
+
+
+# Tag isn't technically an OrganizationalModel, but it has all of its features **except** dynamic-groups
+class TagFactory(OrganizationalModelFactory):
```

Comment:
IMO make this comment a docstring instead so that it carries over in IDEs and stuff.